central/group/datastore/datastore_impl.go

```diff
@@ -213,17 +217,22 @@ func (ds *dataStoreImpl) Remove(ctx context.Context, props *storage.GroupPropert
 		propsID = id
 	}
 
+	err := ds.validateMutableGroupID(ctx, propsID, force)
```

Comment:
Same notion as above, when we create a `validateAndPrepGroupForDeleteNoLock` function we can simplify this here as well.